app/src/store/data.ts

```diff
@@ -48,6 +59,53 @@ export default function useDataStore() {
     lastBlockNumber.value = (blockNumber as BigNumber).toNumber();
     lastBlockTimestamp.value = (timestamp as BigNumber).toNumber();
     grants.value = grantsList as Grant[];
+
+    // Get all rounds from GrantRoundCreated --- TODO: We need to cache these events somewhere (like the graph)
+    const roundList = await roundFactory.value.queryFilter(roundFactory.value.filters.GrantRoundCreated());
+    const roundAddresses = [...roundList.map((e) => e.args?.grantRound)];
+
+    // Pull state from each GrantRound
+    grantRounds.value = await Promise.all(
+      roundAddresses.map(async (grantRoundAddress) => {
+        const round = new Contract(grantRoundAddress, GRANT_ROUND_ABI, provider.value);
+        const donationToken = await round.donationToken();
+        // --- TODO: Cache token contracts and the name/sybmol
+        const token = new Contract(donationToken, ERC20_ABI, provider.value);
+        // get the token details
+        const donationTokenName = await token.name();
+        const donationTokenSymbol = await token.symbol();
+        const donationTokenDecimals = await token.decimals();
+        // get the rounds balance (in whole units)
+        const funds = (await token.balanceOf(grantRoundAddress)) / 10 ** donationTokenDecimals;
+        // discover status from startTime/endTime
+        const now = Date.now();
+        const startTime = await round.startTime();
+        const endTime = await round.endTime();
+        const status =
+          now > startTime.toNumber() * 1000 && now < endTime.toNumber() * 1000
+            ? 'Active'
+            : Date.now() < round.startTime.toNumber() * 1000
+            ? 'Upcoming'
+            : 'Completed';
+
+        return {
+          address: grantRoundAddress,
+          startTime: startTime,
+          endTime: endTime,
+          status: status,
+          donationToken: donationToken,
+          donationTokenName: donationTokenName,
+          donationTokenSymbol: donationTokenSymbol,
+          donationTokenDecimals: donationTokenDecimals,
```

Comment:
My personal preference is always pass around tokens as the [`TokenInfo`](https://github.com/Uniswap/token-lists/blob/fdf9266aefedb9927b182faa8e673a0c29385892/src/types.ts#L1-L12) type defined in Uniswap's [`@uniswap/token-lists`](https://github.com/Uniswap/token-lists) package. It's a nice standard that removes any ambiguity around whether e.g. `donationToken: string;` refers to a symbol, name, or address

In this case we could replace these four fields with:
```suggestion
          donationToken: {
            chainId: <current chainId user is connected to, or 1 otherwise>,
            address: donationToken,
            name: donationTokenName,
            symbol:  donationTokenSymbol,
            decimals: donationTokenDecimals,
            logoURI: null, // TODO we can leave this out for now, and leave a TODO to fetch it from CoinGecko's huge token list (as well as use that to avoid a network request for token info each poll): https://tokenlists.org/token-list?url=https://tokens.coingecko.com/uniswap/all.json
          }
```

Reply:
Resolved in b35f417